src/commands/distribute/releases/edit-notes.ts

```diff
@@ -0,0 +1,117 @@
+
+  @help("Release notes text")
+  @shortName("n")
+  @longName("release-notes")
+  @hasArg
+  public releaseNotes: string;
+
+  @help("Path to release notes file")
+  @shortName("N")
+  @longName("release-notes-file")
+  @hasArg
+  public releaseNotesFile: string;
+
+  public async run(client: AppCenterClient): Promise<CommandResult> {
+    const app = this.app;
+
+    this.validateParameters();
+
+    const releaseId = Number(this.releaseId);
+    if (!Number.isSafeInteger(releaseId) || releaseId <= 0) {
+      return failure(ErrorCodes.InvalidParameter, `${this.releaseId} is not a valid release id`);
+    }
+
+    let releaseDetails: models.ReleaseDetailsResponse;
+    try {
+      debug("Loading release details");
+      const httpRequest = await out.progress(
+        "Loading release details...",
+        clientRequest<models.ReleaseDetailsResponse>((cb) =>
+          client.releasesOperations.getLatestByUser(this.releaseId, app.ownerName, app.appName, cb)
+        )
+      );
+      if (httpRequest.response.statusCode >= 400) {
+        return httpRequest.response.statusCode === 404
+          ? failure(ErrorCodes.InvalidParameter, `release ${this.releaseId} doesn't exist`)
+          : failure(ErrorCodes.Exception, "failed to load release details");
+      } else {
+        releaseDetails = httpRequest.result;
+      }
+    } catch (error) {
+      handleHttpError(error, false, "failed to load release details");
+    }
+
+    const releaseNotes = await this.getReleaseNotesString()
+
+    try {
+      debug(`Updating release notes`);
+      const httpResponse = await out.progress(
+        `Updating release notes`,
+        clientRequest((cb) =>
+          client.releasesOperations.updateDetails(releaseId, app.ownerName, app.appName, { releaseNotes: releaseNotes }, cb)
+        )
+      );
+      if (httpResponse.response.statusCode >= 400) {
+        return failure(ErrorCodes.Exception, `failed to update the relese notes`);
+      }
+    } catch (error) {
+      debug(`Failed to update the release - ${inspect(error)}`);
+      return failure(ErrorCodes.Exception, `failed to update the relese notes`);
+    }
+
+    out.text(`Release notes for ${releaseDetails.shortVersion} (${releaseDetails.version}) with id: ${this.releaseId} has been updated`);
+    return success();
+  }
+
+  private async getReleaseNotesString(): Promise<string> {
+    if (!_.isNil(this.releaseNotesFile)) {
+      try {
+        return await Pfs.readFile(this.releaseNotesFile, "utf8");
+      } catch (error) {
+        if (error.code === "ENOENT") {
+          throw failure(ErrorCodes.InvalidParameter, `release notes file '${this.releaseNotesFile}' doesn't exist`);
+        } else {
+          throw error;
+        }
+      }
+    } else {
+      return this.releaseNotes;
+    }
+  }
+
+  private validateParameters(): void {
+    debug("Checking for invalid parameter combinations");
+    if (!_.isNil(this.releaseNotes) && !_.isNil(this.releaseNotesFile)) {
```

Comment:
What happens when both are unspecified? Note that I don't know the behaviour of the API either.

Reply:
If both unspecified It probably will erase the notes. No way to avoid this issue with empty string (or null paramater).

Reply:
I think the empty string would be explicit (i.e. `--release-notes ""`), as opposed to `undefined` which would be the result of not passing in either of `--release-notes` or `--release-notes-files`.

Reply:
And removing the notes by passing in "" is just fine (assuming it's allowed by the API), but what happens when neither option is passed seems unpredictable.

Reply:
@Vacxe let's make sure that at least one of the two parameters is specified. I tested the command line and specifying no param does not do anything, while specifying `-n ''` resets the notes to nothing (displayed as -).